xpk.py

```diff
@@ -2481,14 +2481,37 @@ def workload_delete(args) -> int:
   if set_cluster_command_code != 0:
     xpk_exit(set_cluster_command_code)
 
-  yml_string = workload_delete_yaml.format(args=args)
-  tmp = write_temporary_file(yml_string)
-  command = f'kubectl delete -f {str(tmp.file.name)}'
-  return_code = run_command_with_updates(command, 'Delete Workload', args)
-
-  if return_code != 0:
-    xpk_print(f'Delete Workload request returned ERROR {return_code}')
-    xpk_exit(return_code)
+  if not args.workload:
+    args.filter_by_status = "EVERYTHING"
+    columns = {
+      'Jobset Name': '.metadata.ownerReferences[0].name',
+    }
+    xpk_print("Get the name of the workloads in the cluster.")
+    return_code, return_value = get_workload_list(args, columns)
+
+    if return_code != 0:
+      xpk_print(f'List Job request returned ERROR {return_code}')
+      xpk_exit(return_code)
+    # Skip the header
+    workloads = return_value.strip().split("\n")[1:]
+  else:
+    workloads = [args.workload]
```

Comment:
Hmm I think we should skip the prompt if they use args.workload since that keeps the current behavior the same. Users are currently happy with the delete single workload case and I don't think we need to doublecheck users there.

Reply:
Gotcha, updated!